File: src/main/java/com/eager/questioncloud/workspace/WorkSpaceQuestionUpdater.java
package com.eager.questioncloud.workspace;

import com.eager.questioncloud.question.Question;
import com.eager.questioncloud.question.QuestionContent;
import com.eager.questioncloud.question.QuestionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class WorkSpaceQuestionUpdater {
    private final QuestionRepository questionRepository;

    public void modify(Long creatorId, Long questionId, QuestionContent questionContent) {
        Question question = questionRepository.getForModifyAndDelete(questionId, creatorId);
        question.modify(questionContent);
        questionRepository.save(question);
    }
}

File: qc-core/qc-workspace/qc-workspace-api/src/main/kotlin/com/eager/questioncloud/workspace/implement/WorkspaceQuestionUpdater.kt
package com.eager.questioncloud.workspace.implement

import com.eager.questioncloud.question.api.internal.ModifyQuestionAPIRequest
import com.eager.questioncloud.question.api.internal.QuestionCommandAPI
import com.eager.questioncloud.workspace.command.ModifyQuestionCommand
import org.springframework.stereotype.Component

@Component
class WorkspaceQuestionUpdater(
    private val questionCommandAPI: QuestionCommandAPI,
) {
    fun modifyQuestion(command: ModifyQuestionCommand) {
        questionCommandAPI.modify(
            ModifyQuestionAPIRequest(
                command.creatorId,
                command.questionId,
                command.questionCategoryId,
                command.subject,
                command.title,
                command.description,
                command.thumbnail,
                command.fileUrl,
                command.explanationUrl,
                command.questionLevel,
                command.price
            )
        )
    }
}